packages/@aws-cdk/core/lib/duration.ts

```diff
@@ -309,12 +309,11 @@ class TimeUnit {
 
 function convert(amount: number, fromUnit: TimeUnit, toUnit: TimeUnit, { integral = true }: TimeConversionOptions) {
   if (fromUnit.inMillis === toUnit.inMillis) { return amount; }
-  const multiplier = fromUnit.inMillis / toUnit.inMillis;
 
   if (Token.isUnresolved(amount)) {
     throw new Error(`Unable to perform time unit conversion on un-resolved token ${amount}.`);
   }
-  const value = amount * multiplier;
+  const value = (amount * fromUnit.inMillis) / toUnit.inMillis;
```

Comment:
Why not a `Math.round()` ?

Reply:
From what I gathered of the usages of this method, they generally want the `Math.floor` of things. Even when not generating a human string. However, the millis to hour conversion when multiplied by 1 hour resulted in the value being represented as 0.99999..... which gets floored to 0.

The insistence on `Math.floor` I think comes from `.toMinutes()` and friends failing if the number isn't an integer so that things like `Duration.seconds(1).toMinutes()` complain.